### DISCUSSION TITLE
what is a "Row Id" in Automation

### DISCUSSION THREAD

**Discussion by shaikhashraf709** [Budibase/budibase#1172] [Help] [answered]
When i try to update a column in automation it asks for "row id" (in the last) what id do i have to put there exactly? and how can i address a particular row in the budibase's internal table?

**Comment by mike12345567** [comment, marked as answer]
Hi @shaikhashraf709 thanks for the question, right now every piece of data stored in a Budibase table has an `ID` - a UUID is used to manage it. In an automation there is really only two ways to get this right now, If the trigger for your automation is either `Row Deleted` or `Row Saved` you'll have access to the row ID as a binding.
If you open the row ID binding menu (using the pencil icon to the right of the input) you'll see the following:
![image](https://user-images.githubusercontent.com/4407001/108840770-b7a9bf80-75ce-11eb-9933-1abcf6fc83cd.png)
The `Id string` can be seen here, which can be input to perform an update to a saved field. This is less useful for a deleted row as it would simply re-create the row, but it is possible for niche use cases.

We still have some work to do on automations, specifically two pieces of work that will make it a lot easier to work with tables:
1. The ability to search for rows based on some input parameters, the next release will introduce the ability to search Budibase tables, in a future release we will be adding an Automation action which can perform a search on a table to get a number of rows.
2. The ability to loop over a number of rows to perform some actions on each one of them. This means you can search for a single row, or every row matching some parameters and execute some actions on them.

This means that to address one particular row in Budibase, one that the automation wasn't triggered on, you'll simply be able to search for it using information from the trigger and then execute actions upon it once its found. This idea is quite similar to the concept of a "Repeater" that we have in the design, the ability to fetch data and then build something out of it.

As I say this is not supported yet and is still being designed, if you'd be interested in features like these please let us know!

**Comment by shaikhashraf709** [comment]
Thanks for looking into it. also the Quick response is appreciable.
